CAP/gap/UniversalObjects.gi

```diff
@@ -421,6 +438,20 @@ InstallMethod( MorphismBetweenDirectSums,
 end );
 
 ##
+InstallMethod( MorphismBetweenDirectSums,
+               [ IsCapCategoryObject, IsList, IsCapCategoryObject ],
+               
+  function( S, morphism_matrix, T )
+    
+    if IsZero( S ) or IsZero( T ) then
```

Comment:
The added implementation of `DirectSumOp` suggests that this is the use-case, and I also think this should be the way to do it. E.g., for modules, @sebastianpos way would produce the morphism one might have wanted to produce, whereas this implementation would construct a matrix made of `0`s.